src/main/java/core/basesyntax/converter/StringToFruitTransactionConverter.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class StringToFruitTransactionConverter implements Function<String, FruitTransaction> {
+
+    @Override
+    public FruitTransaction apply(String s) {
+        StringToFruitValidator validator = new StringToFruitValidator();
+        String[] arr = s.split(",");
+        if (!validator.validateInputData(arr)) {
```

Comment:
The validation logic seems to be in place, but ensure that it covers all edge cases, such as invalid quantity or operation codes. This aligns with the checklist requirement to handle invalid input data properly.